src/goose/job.clj

```diff
@@ -15,9 +15,14 @@
    :retry-opts     retry-opts
    :enqueued-at    (u/epoch-time-ms)})
 
+(defn retried? [job]
+  (if (get-in job [:state :error])
+    true
+    false))
+
 (defn ready-queue
```

Comment:
Shall we rename this to `ready-or-retry-queue`?
It has brought confusion multiple times in past.

Reply:
I will add a comment to rename it to `ready-or-retry-queue`, we can create a different story for this, since it will lead to unrelated changes in the PR.

Reply:
Don't want to create a GH issue just for renaming a function. Name-change refactorings are acceptable in a PR. 

Refactorings around large chunks of logic-modifications/code restructuring require a different story IMO.

@alishamohanty you can take a call on this.

Reply:
I will rename the function after all the functional chunks are over for batching(before the release), also added a todo for the same. Hence, not addressing here.